### PR TITLE
Add Nanocompore

### PR DESCRIPTION
**Pull request by stephanflemming**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Website: https://nanocompore.rna.rocks/
Github: https://github.com/tleonardi/nanocompore
(Bio/Ana)conda: https://anaconda.org/bioconda/nanocompore
Recipe: https://github.com/bioconda/bioconda-recipes/blob/master/recipes/nanocompore/meta.yaml
(My) parameteroverview: https://docs.google.com/spreadsheets/d/1LzZ-ryzfrECsdR3u2AcloFMHFH4y6AxT09mUFSJRuLs/edit#gid=0

Fix: https://github.com/galaxyproject/tools-iuc/issues/2991
Ping: @mmiladi @bgruening

### CODE REVIEW

File: tools/nanocompore/sampcomp.xml [hunk 1]
```diff
@@ -61,40 +68,158 @@
...
+         <data name="out_results" format="tabular" from_work_dir="results/out_nanocompore_results.tsv" label="${tool.name} on ${on_string}: Results">
+            <filter>'results' in ap['out']</filter>
+        </data>
+         <data name="out_shift" format="tabular" from_work_dir="results/out_nanocompore_shift_stats.tsv" label="${tool.name} on ${on_string}: Shift stats">
+            <filter>'shift' in ap['out']</filter>
+        </data>
+        <data name="out_dbbak" format="txt" from_work_dir="results/out_SampComp.db.bak" label="${tool.name} on ${on_string}: Database (bak)">
+            <filter>'dbbak' in ap['out']</filter>
+        </data>
+        <data name="out_dbdat" format="data" from_work_dir="results/out_SampComp.db.dat" label="${tool.name} on ${on_string}: Database (dat)">
```
Comment (bgruening) [inline tools/nanocompore/sampcomp.xml:101]: format="data" is wrong, what is that?

Comment (stephanflemming) [reply tools/nanocompore/sampcomp.xml:101]: A python GDBM object database stored in this binary db.dat

Comment (bgruening) [reply tools/nanocompore/sampcomp.xml:101]: ok, please use format="gdbm". I will add a datatype to Galaxy.

**Comment by stephanflemming** [comment]
Testcase for parameter --bed is missing. 

I am not sure if --allow_warnings should be included --> "Should runtime warnings during the ANOVA tests raise an error?"

**Comment by mmiladi** [comment]
@stephanflemming Thanks a lot for this great push ahead!

Regarding the Nanocompore output, I would think the pragmatist solution could be to tar all the output directory for the downstream analysis or for evaluation within Galaxy Jupyter notebook

**Comment by mmiladi** [comment]
The .tsv files can be reported as standard output I would think. The *.db.dir *.db.dat seems to be needed together for the DB. I also get a *.db.bak file. The last one might also be needed according to here: https://stackoverflow.com/questions/16171833/why-does-the-shelve-module-in-python-sometimes-create-files-with-different-exten

**Comment by stephanflemming** [comment]
Replaced the single output files of the database with a tar containing db.bak, db.dat, and db.dir. Kept the other files as separate result entries.

@mmiladi Fine like that? 
@bgruening gdbm format type is now not needed/used.

**Pull request merged**